src/qilib/utils/storage/memory.py

```diff
@@ -122,3 +131,29 @@ def tag_in_storage(self, tag: TagType) -> bool:
                 return False
         return True
 
+    def load_individual_data(self, tag: TagType, field: Union[str, int, None]) -> Any:
+        """ Retrieve the value of an individual field at the given tag
+
+        Args:
+            tag: The tag
+            field: Name of the field to load
+
+        Returns:
+            Value of the field
+
+        """
+        self._validate_tag(tag)
+        return self._unserialize(StorageMemory._retrieve_value_from_dict_by_tag(self._data, tag, field))
+
+    def update_individual_data(self, data: Any, tag: TagType, field: Union[str, int, None]) -> None:
```

Comment:
Union[str, int, None] -> Optional[Union[str, int]]

Reply:
Only applying type hinting doesn't prevent errors when the user uses something different than int, str. Maybe check type of field and raising descriptive TypeError when not str, int